metaflow/metaflow_config_funcs.py

```diff
@@ -0,0 +1,128 @@
+import json
+import os
+
+from collections import namedtuple
+
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+ConfigValue = namedtuple("ConfigValue", "value serializer is_propagate is_default")
+
+NON_CHANGED_VALUES = 1
+NON_PROPAGATED_VALUES = 2
+NULL_VALUES = 4
+ALL_VALUES = 7
+
+
+def init_config():
+    # Read configuration from $METAFLOW_HOME/config_<profile>.json.
+    home = os.environ.get("METAFLOW_HOME", "~/.metaflowconfig")
+    profile = os.environ.get("METAFLOW_PROFILE")
+    path_to_config = os.path.join(home, "config.json")
+    if profile:
+        path_to_config = os.path.join(home, "config_%s.json" % profile)
+    path_to_config = os.path.expanduser(path_to_config)
+    config = {}
+    if os.path.exists(path_to_config):
+        with open(path_to_config, encoding="utf-8") as f:
+            return json.load(f)
+    elif profile:
+        raise MetaflowException(
+            "Unable to locate METAFLOW_PROFILE '%s' in '%s')" % (profile, home)
+        )
+    return config
+
+
+# Initialize defaults required to setup environment variables.
+METAFLOW_CONFIG = init_config()
+
+_all_configs = {}
+
+
+def config_values(include=0):
+    # By default, we just return non-null values that are marked as "propagate" and that
+    # are not default. This is the common use case because in all other cases, the code
+    # is sufficient to recreate the value (ie: there is no external source for the value)
+    for name, config_value in _all_configs.items():
+        if (
+            (config_value.value is not None or include & NULL_VALUES)
+            and (config_value.is_propagate or include & NON_PROPAGATED_VALUES)
+            and (not config_value.is_default or include & NON_CHANGED_VALUES)
+        ):
+            yield name, config_value.serializer(config_value.value)
```

Comment:
Will this function be used by downstream integrations like `batch`, `sfn`, `argo`,`airflow` etc to just get all the config values for metaflow? That seems like a very good step forward! It makes any new external remote integration so much easier!.

Reply:
If you want it to be. I have not included it in any of the OSS integrations (`batch`, `sfn`, `argo`, `airflow`) because @savingoyal mentioned it would take a *lot* longer to test. I am using this internally for exactly the reason you specify (I am guaranteed that all configuration values are passed remotely and if a new one shows up, it is automatically passed along). But again, it's up to you if you want to use it or not. This PR does not make that change.